src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -2090,7 +2090,7 @@ private BoundExpression BindNonMethod(SimpleNameSyntax node, Symbol symbol, Bind
                         }
                         else if (parameter.IsExtensionParameter() &&
                                 (InParameterDefaultValue || InAttributeArgument ||
-                                 this.ContainingMember() is not { Kind: not SymbolKind.NamedType, IsStatic: false } || // We are not in an instance member
+                                 this.ContainingMember() is { Kind: SymbolKind.NamedType } or { IsStatic: true } || // We are not in an instance member
```

Comment:
Strictly speaking this rewrite is not semantically equivalent because the recursive pattern includes an implicit `not null` test. Therefore, the process of pushing the `not` down the pattern should result in an explicit `null` test, combined with `or` with the rest. It is quite possible that it was intentional to have the implicit `not null` test to protect the following line from a possible null dereference. I suggest to use semantically equivalent rewrite and to protect the recently added ```this.ContainingMember().IsStatic``` access in the body of the `if`